src/nvim/tui/termkey/termkey.c

```diff
@@ -637,34 +637,7 @@ static void eat_bytes(TermKey *tk, size_t count)
 // TODO(dundargoc): we should be able to replace this with utf_char2bytes from mbyte.c
 int fill_utf8(int codepoint, char *str)
 {
-  int nbytes = utf_char2len(codepoint);
-
-  str[nbytes] = 0;
-
-  // This is easier done backwards
-  int b = nbytes;
-  while (b > 1) {
-    b--;
-    str[b] = (char)0x80 | (codepoint & 0x3f);
-    codepoint >>= 6;
-  }
-
-  switch (nbytes) {
```

Comment:
I guess we could replace the implementation in `utf_char2bytes` with this as they do the same thing, but don't think it's a big readability win either way. Can replace if anyone likes this version more.